tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts

```diff
@@ -166,15 +166,37 @@ export function getFilteredParent(apiItem: ApiItem): ApiItem | undefined {
 }
 
 /**
- * Adjusts the name of the item as needed.
- * Accounts for method overloads by adding a suffix such as "myMethod_2".
+ * Converts invalid filename characters to underscores.
+ */
+function getFileSafeNameForName(apiItemName: string): string {
+	// eslint-disable-next-line unicorn/better-regex, no-useless-escape
+	const badFilenameCharsRegExp: RegExp = /[^a-z0-9_\-\.]/gi;
+
+	// Note: This can introduce naming collisions.
+	// TODO: once the following issue has been resolved in api-extractor, we may be able to clean this up:
+	// https://github.com/microsoft/rushstack/issues/1308
+	return apiItemName.replace(badFilenameCharsRegExp, "_").toLowerCase();
+}
+
+/**
+ * Gets a filename-safe representation of the API item's display name.
+ *
+ * @remarks
+ * - Handles invalid filename characters.
+ *
+ * - Accounts for method overloads by adding a suffix (such as "myMethod_2").
  *
  * @param apiItem - The API item for which the qualified name is being queried.
  *
  * @public
  */
-export function getQualifiedApiItemName(apiItem: ApiItem): string {
-	let qualifiedName: string = getSafeFilenameForName(apiItem.displayName);
+export function getFileSafeNameForApiItem(apiItem: ApiItem): string {
+	const unqualifiedName: string =
+		apiItem.kind === ApiItemKind.Package
+			? getFileSafeNameForName(getUnscopedPackageName(apiItem as ApiPackage))
```

Comment:
Curious if there was a big reason why we unscope the package name. Feels like at least in some cases the scope could still be useful in the filename-safe output of the function.

Reply:
Huh. You know, that's fair. We would still need to omit the `/` between the scope and the package name in those cases, but the `getFileSafeNameForName` helper would handle that for us.

I still think the _default_ behavior probably wants to be to omit the scope, but there's no reason to enforce that. I'll update this.

Reply:
Updated